From faba5d52ae82239f522c108b51fa18dbb1ae9a8b Mon Sep 17 00:00:00 2001
From: Mark Lodato <lodatom@gmail.com>
Date: Mon, 5 Oct 2009 22:49:07 -0400
Subject: [PATCH] fix -Wextra warning for empty `else' body

The macros XDECREF, XGIVEREF, and XGOTREF used ";" to denote an empty if
and else body, but these raised the following warning when compiled with
"gcc -Wall -Wextra":

    warning: suggest braces around empty body in an `else' statement
---
 Cython/Compiler/ModuleNode.py |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Cython/Compiler/ModuleNode.py b/Cython/Compiler/ModuleNode.py
index 5fc77ca..0a396c4 100644
--- a/Cython/Compiler/ModuleNode.py
+++ b/Cython/Compiler/ModuleNode.py
@@ -2472,7 +2472,7 @@ static __Pyx_RefnannyAPIStruct *__Pyx_Refnanny = NULL;
 #define __Pyx_DECREF(r) __Pyx_Refnanny->DECREF(__pyx_refchk, (PyObject *)(r), __LINE__)
 #define __Pyx_GOTREF(r) __Pyx_Refnanny->GOTREF(__pyx_refchk, (PyObject *)(r), __LINE__)
 #define __Pyx_GIVEREF(r) __Pyx_Refnanny->GIVEREF(__pyx_refchk, (PyObject *)(r), __LINE__)
-#define __Pyx_XDECREF(r) if((r) == NULL) ; else __Pyx_DECREF(r)
+#define __Pyx_XDECREF(r) do { if((r) != NULL) {__Pyx_DECREF(r);} } while(0)
 #define __Pyx_SetupRefcountContext(name) \
   void* __pyx_refchk = __Pyx_Refnanny->NewContext((name), __LINE__, __FILE__)
 #define __Pyx_FinishRefcountContext() \
@@ -2486,8 +2486,8 @@ static __Pyx_RefnannyAPIStruct *__Pyx_Refnanny = NULL;
 #define __Pyx_SetupRefcountContext(name)
 #define __Pyx_FinishRefcountContext()
 #endif /* CYTHON_REFNANNY */
-#define __Pyx_XGIVEREF(r) if((r) == NULL) ; else __Pyx_GIVEREF(r)
-#define __Pyx_XGOTREF(r) if((r) == NULL) ; else __Pyx_GOTREF(r)
+#define __Pyx_XGIVEREF(r) do { if((r) != NULL) {__Pyx_GIVEREF(r);} } while(0)
+#define __Pyx_XGOTREF(r) do { if((r) != NULL) {__Pyx_GOTREF(r);} } while(0)
 """)
 
 
-- 
1.6.5.rc1

